smartapps/bangali/rooms-child-app.src/rooms-child-app.groovy

```diff
@@ -2410,27 +2413,165 @@ def processCoolHeat()       {
 
 def processCoolHeat()       {
     ifDebug("processCoolHeat")
-    def temp = -1
-    def child = getChildDevice(getRoom())
+       
+    //Need to stop things even if nobody's in
     def isHere = (personsPresence ? personsPresence.currentValue("presence").contains('present') : false)
     if ((checkPresence && !isHere) || maintainRoomTemp == '4')    {
-        if (checkPresence && !isHere)       {
-            if (['1', '3'].contains(maintainRoomTemp))
-                (useThermostat ? roomThermostat.off() : roomCoolSwitch.off())
-            if (['2', '3'].contains(maintainRoomTemp))
-                (useThermostat ? roomThermostat.off() : roomHeatSwitch.off())
-        }
         updateMaintainIndP(temp)
         updateThermostatIndP(isHere)
-        return
     }
-    def roomState = child?.currentValue('occupancy')
+    if (maintainRoomTemp == '4')
+    	return
+       
     def temperature = getAvgTemperature()
     def updateMaintainIndicator = true
-    def turnOn = null
-    def thisRule = [:]
-    if (state.rules)    {
+    
+    def turnOn = getActiveTemperatureRule()   
+	ifDebug("processCoolHeat: rule: $turnOn")
+    if (turnOn)     {
+        def thisRule = getRule(turnOn, 't')
+        def tempRange = thisRule.tempRange
+		def setpoint       
+        //reduced the off commands and fan commands to prevent spamming the likes of Nest, which have an API rate limiter
```

Comment:
state variables added to prevent messing with the thermostat if not needed

Reply:
why the state variable instead of checking the current value of the thermostat to determine if heating or cooling is active?

thanks.

Reply:
For activating, the state variables are redundant since we quit if HVAC is running (and I should remove the checks really), but for cooling it stops us from turning off activations from other rooms and from the thermostat itself.

Reply:
will make the response to this a part of the centralized model discussion above.